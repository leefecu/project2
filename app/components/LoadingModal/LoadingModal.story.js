import React from 'react';
import { Text, View } from 'react-native';

import { storiesOf } from '@storybook/react-native';

import LoadingModal from './LoadingModal';
import Styles from '../../Stories.styles';

storiesOf('Component | LoadingModal', module).add('Primary', () => {
  return (
    <View style={Styles.containerWithBG}>
      <View style={Styles.row}>
        <Text style={Styles.title}>HUH</Text>
        <LoadingModal />
      </View>
      <View style={Styles.row}>
        <Text style={Styles.title}>OOPS</Text>
        <LoadingModal />
      </View>
    </View>
  );
});
