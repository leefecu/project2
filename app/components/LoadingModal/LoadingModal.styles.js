import { StyleSheet } from 'react-native';

import Colors from '../../themes/Colors';
import Metrics from '../../themes/Metrics';

const SMALL_DEVICE = Metrics.width <= 320; // iphone 5

export default StyleSheet.create({
  header: {
    position: 'relative',
    alignItems: 'center',
  },
  headerLine: {
    position: 'absolute',
    backgroundColor: Colors.primaryDark,
    height: 1,
    top: 9,
    width: Metrics.width - Metrics.doubleMargin,
  },
  headerTextWrapper: {
    backgroundColor: Colors.white,
    paddingHorizontal: Metrics.doublePadding,
    zIndex: 1,
  },
  headerText: {
    color: Colors.primaryDark,
    fontSize: Metrics.Fonts.small,
    fontFamily: 'CircularPro-Bold',
    fontWeight: 'bold',
    textAlign: 'center',
  },
  content: {
    marginVertical: SMALL_DEVICE ? Metrics.doubleMargin * 2 : Metrics.doubleMargin * 2,
  },
  message: {
    color: Colors.primaryDark,
    fontSize: SMALL_DEVICE ? Metrics.Fonts.small : Metrics.Fonts.normal,
    fontFamily: 'CircularPro-Book',
    marginBottom: SMALL_DEVICE ? Metrics.smallMargin : Metrics.doubleMargin,
    textAlign: 'center',
  },
  messageContainer: {
    marginBottom: Metrics.doubleMargin,
    paddingHorizontal: Metrics.basePadding,
  },
  loader: {
    alignItems: 'center',
  },
});
