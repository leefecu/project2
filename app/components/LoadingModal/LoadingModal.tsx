import React from 'react';
import { ActivityIndicator, Text, View } from 'react-native';

import { testProps } from '../../helpers/general';

import Styles from './LoadingModal.styles';

import Colors from '../../themes/Colors';

const LoadingModal = () => (
  <View {...testProps('Loading Modal')}>
    <View style={Styles.header}>
      <View style={Styles.headerLine} />
      <View style={Styles.headerTextWrapper}>
        <Text style={Styles.headerText}>LOADING</Text>
      </View>
    </View>

    <View style={Styles.content}>
      <View style={Styles.messageContainer}>
        <Text style={Styles.message}>Weâ€™re processing your request.</Text>
      </View>

      <View style={Styles.loader}>
        <ActivityIndicator size="large" color={Colors.loadingBlue} />
      </View>
    </View>
  </View>
);

export default LoadingModal;
