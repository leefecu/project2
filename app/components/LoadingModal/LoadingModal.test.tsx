/**
 * @jest-environment jsdom
 */
import React from 'react';
import { create } from 'react-test-renderer';

import LoadingModal from './LoadingModal';

describe('LoadingModal component', () => {
  test('render default LoadingModal', () => {
    const button = create(<LoadingModal />);
    expect(button.toJSON()).toMatchSnapshot();
  });
});
