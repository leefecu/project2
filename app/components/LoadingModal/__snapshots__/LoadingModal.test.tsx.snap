// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingModal component render default LoadingModal 1`] = `
<View
  accessibilityLabel="Loading Modal"
  testID="Loading Modal"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "position": "relative",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#004071",
          "height": 1,
          "position": "absolute",
          "top": 9,
          "width": 718,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "paddingHorizontal": 24,
          "zIndex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#004071",
            "fontFamily": "CircularPro-Bold",
            "fontSize": 17,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        LOADING
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 64,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 32,
          "paddingHorizontal": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#004071",
            "fontFamily": "CircularPro-Book",
            "fontSize": 20,
            "marginBottom": 32,
            "textAlign": "center",
          }
        }
      >
        Weâ€™re processing your request.
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#0faae5"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
  </View>
</View>
`;
